darts/tests/test_timeseries_static_covariates.py

```diff
@@ -154,6 +154,42 @@ def test_from_group_dataframe(self):
         assert (ts[0].values().flatten() == [values[2], values[1], values[0]]).all()
         assert (ts[1].values().flatten() == [values[3], values[4], values[5]]).all()
 
+        # for time as string dates
```

Comment:
I would take this test as a way to get familiar with test parametrization (e.g. writing a generic test that can runs/test on collection of inputs).

You would need to adapt the test so that it expects an input time index type, and from that it will generate the corresponding index and do the entire logic only once.

You can then use the `@pytest.mark.parametrize(...)` decorator to run different test configurations.

Spoiler: Below is an example of how this test could be simplified:

```
    @pytest.mark.parametrize(
        "index_type",
        ["int", "dt", "str"]
    )
    def test_from_group_dataframe(self, index_type):
        """Tests correct extract of TimeSeries groups from a long DataFrame with unsorted (time/integer) index"""
        group = ["a", "a", "a", "b", "b", "b"]
        values = np.arange(len(group))

        # time as an unsorted list
        if index_type == "int":
            index_expected = pd.RangeIndex(3)
            time = [2, 1, 0, 0, 1, 2]
        else:  # datetime index
            index_expected = pd.date_range("2024-01-01", periods=3)
            time = index_expected[::-1].append(index_expected)
            if index_type == "str":
                time = time.astype(str)

        # for time as a unsorted list
        df = pd.DataFrame({
            "group": group,
            "time": time,
            "x": values,
        })
        ts = TimeSeries.from_group_dataframe(df, group_cols="group", time_col="time")

        # check the time index
        assert ts[0].time_index.equals(index_expected)
        assert ts[1].time_index.equals(index_expected)

        # check the values
        assert (ts[0].values().flatten() == [values[2], values[1], values[0]]).all()
        assert (ts[1].values().flatten() == [values[3], values[4], values[5]]).all()
```

Reply:
thanks (done so in the latest commit)